src/main/features/ccj/routes/payment-options.ts

```diff
@@ -27,6 +27,9 @@ export default express.Router()
           res.render(Paths.paymentOptionsPage.associatedView, { form: form })
         } else {
           user.ccjDraft.paymentOption = form.model
+          if (form.model.option === PaymentType.IMMEDIATELY) {
+            user.ccjDraft.repaymentPlan = user.ccjDraft.payBySetDate = undefined
+          }
```

Comment:
Any reason why other options are cleared only in this case? E.g. repayment plan could be cleared if pay by set date is selected.

Reply:
For other cases we have another page where we clean it. `IMMEDIATELY` redirect to C'nS page.

Reply:
👍